### DISCUSSION TITLE
The CodeProject.AI-Server Discussions

### DISCUSSION THREAD

**Discussion by ChrisMaunder** [codeproject/CodeProject.AI-Server#132] [Announcements]
> [!NOTE]
> Please remember that this project is being maintained by volunteers. Please be patient if responses are slow, or bug fixes and new features aren't implemented as fast as one would hope.

## 👋 Welcome!

CodeProject.AI Server has moved homebase from CodeProject.com to [codeproject.github.io/](https://codeproject.github.io/) to centralise the code, the discussions, and the docs into one self-contained place dedicated to the CodeProject.AI Server product.

This discussion board should focus on the CodeProject.AI Server and installation process only. For all questions on individual modules, please refer to that module's issues board directly. A list of all modules is available at [codeproject.github.io/](https://codeproject.github.io/).

Please use this board to:

* 🙋 Ask general questions, or get help or guidance
* 💡 Make suggestions on features or the overall experience
* 💬 Connect with the community

For bugs, please use Issue board. 🐞

**Comment by Gordohat** [comment]
I have an NVIDIA GeForce GTX 1060 card in my Windows 10 machine.   It is a gen 7 Intel core i7 processor.

Should I be able to just run the latest of all the CUDA stuff, including the CUDA drivers, CUDA Toolkit, cuDNN together with the latest NVIDIA graphics and the Microsoft .NET stuff?   I am only using CP.AI for use with BlueIris.  

If it only will work with specific older versions of the above items, why is this so?  I would like to know what is limiting CP.AI from utilising the latest versions and could it be fixed.

**Reply by ChrisMaunder** [reply]
CodeProject.AI modules are each separate projects so the accurate answer is "it depends on the module", but generally, if your card work with the latest drivers and latest CUDA, then the modules will work too. For the modules we've created we try and ensure they have support for the latest CUDA, but sometimes some things do throw a curve ball, such as the recent NumPy update

**Comment by Gordohat** [comment]
Thanks.  I just need basic YOLO object detection (YOLOv5 6.2 module) for use in Blue Iris on Windows 10.  The CP.AI server is running on the same Windows 10 machine.  Does that help to be more specific in your answer?

**Reply by MikeLud** [reply]
You can also use the Object Detection (YOLOv5 .NET) module. This module has the same models as the Object Detection (YOLOv5 6.2) and does not require CUDA to use your GPU

**Reply by JuLeXoGame** [reply]
> You can also use the Object Detection (YOLOv5 .NET) module. This module has the same models as the Object Detection (YOLOv5 6.2) and does not require CUDA to use your GPU

bu if he has cuda, better to use it if he need to save some ressources on cpu

**Comment by BeanBagKing** [comment]
@ChrisMaunder - There's been some discussion on the future of CodeProject.AI over on the other message boards (for reference: https://www.codeproject.com/Messages/6020926/Re-Code-Project-AI-Server-version-2-8-0). I think the worry is that, with your Aug 31st announcement and with Matthew Dennis no longer involved, that the project may cease to be updated. However, I still see activity from you (this discussion, case in point) over here.

Do you mind commenting on what you think the future of CodeProject.AI looks like at this time? There's a large user base (I feel like anyway) that uses this for Blue Iris, and if there aren't any currently active devs, it would be good to have some forewarning. I know that, as an all-volunteer project, any future plans are subject to sudden change, I'm just trying to get a feel for what's going on. Thank you for all you've contributed over the years, regardless of what the future holds.

**Reply by ChrisMaunder** [reply]
@matthewDDennis and I will still be contributing for as long as we can. This project is close to our heart, and while we're no longer with CodeProject, we're still able to contribute to the project.

**Comment by thecamaroguy** [comment]
Since moving here, will the Docker Images be updated and will they be available at the same location ->   https://hub.docker.com/r/codeproject/ai-server/tags?page=1&name= or will that move as well?

**Reply by matthewDDennis** [reply]
It looks like we still have access to the repositories, so they should get updated.
If we must move them, we will update the documentation and the readme file with the latest information.

**Reply by jameson71** [reply]
> It looks like we still have access to the repositories, so they should get updated. If we must move them, we will update the documentation and the readme file with the latest information.

FYI it appears to me that the :latest tag on docker hub still points to 2.6.5 even though 2.8.0 has been up for about 2 months.

**Reply by ChrisMaunder** [reply]
2.9.X are starting to come online as I get a chance to build and test

**Comment by Armand29** [comment]
When I went into the server, the service was stopped and the "Install Modules" page was blank. I tried to install 2.8.0 (I downloaded it last week, but the download site doesn't work today) and the page came up, told me I couldn't reach the server, then shut down. Is codeproject.ai dead?

**Reply by hspindel** [reply]
> When I went into the server, the service was stopped and the "Install Modules" page was blank. I tried to install 2.8.0 (I downloaded it last week, but the download site doesn't work today) and the page came up, told me I couldn't reach the server, then shut down. Is codeproject.ai dead?

Same here.  Codeproject.ai website seems to be down for maintenance.  Adopting a patient attitude while this gets worked out, although it is frustrating that AI has stopped working for my Blue Iris installation.

**Reply by ChrisMaunder** [reply]
Should be fixed now

**Comment by fobrs** [comment]
Are there any plans to support the Hexagon NPU on the Snapdragon X Elite processors?

**Reply by ChrisMaunder** [reply]
Hexagon - not at the moment. Snapdragon: My general comment is that we support arm64 on Linux, macOS and Windows, so many things will work on the new arm chips.

Until I manage to find a shiny new Snapdragon arm64 machine lying around needing a home, my only way of testing arm64 for Windows is via Windows on Arm using VMWare on my M1 mac mini.

**Comment by mrkilvo** [comment]
Hi All,

I have managed to get this working for Blue Iris, i am sure you are all as frustrated after uninstalling and realising its all messed up and not working, i have only tested this on PC and have Yolov5 3, ALPR and Face Recognition working. 

Here is how I got it working

Install CodeProject 
Download the Module(s) from the webpage

Unzip and copy the module(s) to the following folder c:\Program Files\CodeProject\AI\modules

run a command prompt as Administrator

Navigate to the Directory c:\Program Files\CodeProject\AI\modules\[modulename]

Type ..\..\setup.bat

Restart codeproject



at least it should get it working for now

**Reply by ChrisMaunder** [reply]
The install system should be back working as normal now.

**Comment by pbradley0** [comment]
I'm having trouble adding new modules now.  I'm running the server in docker and have cloned the module repository and then ran the setup.sh from inside the docker image and it completes without error.  The module shows up in the list of available modules on the server webpage but its status is listed as "Not Available."  I've tested this with YOLOv8 and the Coral module and they are both doing it.  Anyone else having this issue?  Am I missing a step?

**Reply by thecamaroguy** [reply]
What is the command you used to create your docker and pull request?

**Reply by MikeLud** [reply]
The below link should have what you are looking for.

https://codeproject.github.io/codeproject.ai/install/running_in_docker.html

**Comment by kmjsmith** [comment]
yeh im having an issue getting python installed as seen in the photo 
![Screenshot 2024-10-28 221338](https://github.com/user-attachments/assets/c0e51281-cea8-4169-80ce-3f4d73b20f01)

**Reply by kmjsmith** [reply]
![Screenshot 2024-10-28 223651](https://github.com/user-attachments/assets/bca1bcc4-291c-4e7f-8827-b711194b11d1)

**Reply by thecamaroguy** [reply]
Under the Appsettings.json you will find:

 "ModuleInstallTimeout": "00:20:00", around line 92 or something.
 
 Increase that value.  I put it to 2 hours, but depending on your internet speeds you might want to increase it to like 20 hours
 "ModuleInstallTimeout": "20:20:00",

**Reply by kmjsmith** [reply]
well it seems to install but it gives the error as seen in my second image even though i have cuda 12.0 installed

**Reply by thecamaroguy** [reply]
Mine did this and appeared to have installed, but always gave this error.  I worked with Chris via the forums and he suggested I remove the module and then adjust this value and reinstall the module.  After doing that, it all worked for me and I have not had issue.  When a new docker comes out, I pull it down, modify that line, then upgrade any modules.  Been flawless since.

**Reply by ChrisMaunder** [reply]
@kmjsmith There should be an install.log file in the directory for that module (or click the "info" button on the module on the dashboard). If you can paste the install log I can make some guess.

**Reply by JuLeXoGame** [reply]
when i see i'm always getting this error and it exist since october 2024, then that mean codeproject is just dead and they dont care

**Reply by ChrisMaunder** [reply]
@JuLeXoGame There's a difference between "not caring" and "having zero time to work on code voluntarily due to current work pressures". I understand your frustration and we all wish we had 48hrs each day, but you have a choice: dive in and offer your own time and expertise to help fix bugs on this Open Source project, or move to another product that suits your needs better.

**Reply by JuLeXoGame** [reply]
i spend hours for finding potentially this working solution, if it's this ridiculous value, and you cant do an update with this value updated, what can i say...

**Comment by hspindel** [comment]
General question from Code Project AI newbie:

I am using CodeProject (server version 2.6.5) with Blue Iris.  Nvidia Quadro P620 GPU.  Windows 11.  The YOLOv5.3.1 module is working well for me.  The install automatically provided ipcam-animal, ipcam-combined, ipcam-dark, ipcam-general, and license-plate.

Thought I would try YOLOv8.  It installs fine and works, but the install only provides ipcam-general and license-plate.

Are the additional modules available for YOLOv8?  Can I just copy them over from YOLOv5.3.1?

Would there be any benefit to me from using YOLOv8?

**Reply by MikeLud** [reply]
Are the additional modules available for YOLOv8? **No**

Can I just copy them over from YOLOv5.3.1? **No**

Would there be any benefit to me from using YOLOv8? **I found there is not much benefit over the other YOLO modules**

**Reply by hspindel** [reply]
Thank you very much for the thorough reply!

**Reply by JuLeXoGame** [reply]
> Are the additional modules available for YOLOv8? **No**
> 
> Can I just copy them over from YOLOv5.3.1? **No**
> 
> Would there be any benefit to me from using YOLOv8? **I found there is not much benefit over the other YOLO modules**

the benefits i could see is being potentially able to save faces? sou dont have false positive trigger

**Comment by hspindel** [comment]
CodeProject (server version 2.6.5) with Blue Iris. Nvidia Quadro P620 GPU. Windows 11. YOLOv5.3.1.

Don't know if this is a new problem, but I just noticed in the logs a complaint that custom-models\actionnetv2.pt does not exist.  I verified that this .pt file actually does not exist in the specified directory.

Is this a problem, or should I ignore it?  AI object recognition works fine.  Is there a change I can make to a config file to tell it not to look for actionnetv2.pt or a way to add it?

What does actionnetv2.pt do?

Thank you.

**Reply by tbc725** [reply]
I think the actionnetv2.pt is for action detection. Like if the person is standing or sitting, etc.

**Reply by hspindel** [reply]
Thank you, that's good to know.

**Comment by thecamaroguy** [comment]
Started to get the following - "Unable to check for updates" on my Docker environment.  2.8.0

It's probably because of the move, but just curious if there is anything I can change to make it point to a new location (if there is one)

**Reply by ChrisMaunder** [reply]
How about now?

**Comment by DeFlanko** [comment]
@ChrisMaunder  as it pertains to Docker Images -- [is this the best way](https://hub.docker.com/r/codeproject/ai-server/tags?name=cuda12) to track the latest Cuda12 builds for docker?

**Reply by ChrisMaunder** [reply]
Yep!

**Comment by fotingo** [comment]
Hello,
I am on version 2.6.5 and ever since then, I get "Unable to check for Updates error"
![image](https://github.com/user-attachments/assets/6246cb0e-6b45-4a73-990c-7f25d58e7341)

I also notice a lot of false positives, like detecting "person" when it's just a shadow. That started happening after the update.
Which version should I get, 2.8 or 2.9.3? 

Is 2.8 the stable version and 2.9.3 beta?

**Comment by thecamaroguy** [comment]
Thank you for updating the docker images to 2.9.4.  So far, everything looks great!

**Reply by thecamaroguy** [reply]
Well, maybe I was too quick:

when downloading Yolov8 Models...I get the following:
Preparing to download model 'objectdetection-coco-yolov8-pt-nsmlx.zip' for module ObjectDetectionYOLOv8
15:07:47:Downloading model 'objectdetection-coco-yolov8-pt-nsmlx.zip' to '/app/downloads/modules/packages/ObjectDetectionYOLOv8/objectdetection-coco-yolov8-pt-nsmlx.zip'
15:07:53:Preparing to download model 'objectsegmentation-coco-yolov8-pt-nsmlx.zip' for module ObjectDetectionYOLOv8
15:07:53:Downloading model 'objectsegmentation-coco-yolov8-pt-nsmlx.zip' to '/app/downloads/modules/packages/ObjectDetectionYOLOv8/objectsegmentation-coco-yolov8-pt-nsmlx.zip'
15:07:56:Preparing to download model 'objectdetection-custom-yolov8-pt-m.zip' for module ObjectDetectionYOLOv8
15:07:56:Downloading model 'objectdetection-custom-yolov8-pt-m.zip' to '/app/downloads/modules/packages/ObjectDetectionYOLOv8/objectdetection-custom-yolov8-pt-m.zip'
15:08:06:Unable to download and install objectdetection-custom-yolov8-pt-m.zip: Unable to download 'objectdetection-custom-yolov8-pt-m.zip'. Error: Could not find a part of the path '/app/downloads/modules/packages/ObjectDetectionYOLOv8/objectdetection-custom-yolov8-pt-m.zip'.
15:08:17:Unable to download and install objectdetection-coco-yolov8-pt-nsmlx.zip: Response timeout. Try increasing the timeout value
15:08:23:Unable to download and install objectsegmentation-coco-yolov8-pt-nsmlx.zip: Response timeout. Try increasing the timeout value

What timeout do I need to increase?

**Reply by ChrisMaunder** [reply]
Are you downloading from the dashboard UI? In this case you can't adjust the timeout (though I will add a TODO to change this).

Generally, timeouts like this are due to our CDN. They are great when they are great. Sometimes, though, they just timeout.

**Comment by christopher-init** [comment]
Hello,

Is there somewhere I can retrieve detailed logs?

I have CPAI running in the latest Ubuntu Server using the latest docker image (GPU support). GPU passthrough is setup and working. CPAI sees the GPU and everything is working as expected.

I use YOLOv5 as it has been the most accurate. I use Medium models, and try to turn off half precision.

The GPU I am using is a GTX 1650 Low Profile.

Now the issue — the module randomly stops. Sometimes “is shutting down” is in the logs and sometimes it is not. I have no clue why, there is no good information in either /app/logs or docker logs. My current workaround which sucks, is using the API to try and “detect” an object using a static image every minute, if it returns a timeout, I run docker restart.

Something else I noticed: 
Half Precision setting in the docker module settings doesn’t work.
No API call to get the latest log entry (would help detect “shutdown”)

Anyway, any help in the right direction would be appreciated.

**Reply by JuLeXoGame** [reply]
if you turn off half precision i guess it is in FP32 instead of FP16, and it demand a lot of more VRAM and ressources, i don't really think it's possible on your gpu
